public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts

```diff
@@ -54,9 +55,24 @@ export default class ResourcePickerData extends DataSourceWithBackend<AzureMonit
         | order by subscriptionURI asc
     `;
 
-    const response = await this.makeResourceGraphRequest<RawAzureResourceGroupItem[]>(query);
+    let resourceResponse = await this.makeResourceGraphRequest<RawAzureResourceGroupItem[]>(query);
+    if (!resourceResponse.data.length) {
+      throw new Error('unable to fetch resource details');
+    }
+    let resources = resourceResponse.data;
+
+    while (resourceResponse.$skipToken) {
+      // The response include several pages
+      resourceResponse = await this.makeResourceGraphRequest<RawAzureResourceGroupItem[]>(query, 1, {
+        $skipToken: resourceResponse.$skipToken,
+      });
+      if (!resourceResponse.data.length) {
+        throw new Error('unable to fetch resource details');
+      }
+      resources = resources.concat(resourceResponse.data);
+    }
```

Comment:
maybe we could make this a single loop? something like

```typescript
let fetchMore = true;

while(fetchMore) {
  const response = ...;
  fetchMore = response.$skipToken

```
or something else fancier? Thoughts?